roofit/hs3/src/JSONIOUtils.cxx

```diff
@@ -1,5 +1,58 @@
 #include "JSONIOUtils.h"
+using namespace nlohmann;
 
-bool endsWith(std::string_view str, std::string_view suffix) {
+bool endsWith(std::string_view str, std::string_view suffix)
+{
    return str.size() >= suffix.size() && 0 == str.compare(str.size() - suffix.size(), suffix.size(), suffix);
-} 
\ No newline at end of file
+}
+
+std::string varJSONString(const std::string &jsonInfoString)
+{
+   json jsonInfoDict = json::parse(jsonInfoString);
+   std::string varName = jsonInfoDict["name"];
+
+   double val = 0;
+   double maxVal = 0;
+   double minVal = 0;
+   bool isConstant = false;
+   bool isRange = false;
+
+   if (jsonInfoDict.contains("value")) {
+      val = jsonInfoDict["value"];
+      isConstant = true;
+   }
+   if (jsonInfoDict.contains("max") && jsonInfoDict.contains("min")) {
+      maxVal = jsonInfoDict["max"];
+      minVal = jsonInfoDict["min"];
+      isRange = true;
+   }
+   if (!isConstant) {
+      val = (maxVal + minVal) / 2;
+   }
+   // Check if variable is at least a range or constant else throw error
+   if (isConstant || isRange) {
```

Comment:
```suggestion
   if (!(isConstant || isRange)) {
      throw std::invalid_argument("Invalid Syntax: Please provide either 'value' or 'min' and 'max' or both");
   }
```
Then the rest of this function doesn't need to be indented and it's easier to read the control flow.

Reply:
This went unnoticed; my bad. I've changed it now.